libs/full/segmented_algorithms/include/hpx/parallel/segmented_algorithms/for_each.hpp

```diff
@@ -287,4 +254,68 @@ namespace hpx { namespace segmented {
             std::forward<ExPolicy>(policy), first, last, std::forward<F>(f),
             hpx::parallel::util::projection_identity(), is_seq());
     }
+
+    // clang-format off
+    template <typename InIter, typename Size,
+        typename F,
+        HPX_CONCEPT_REQUIRES_(
+            hpx::traits::is_iterator<InIter>::value &&
+            hpx::traits::is_segmented_iterator<InIter>::value
+        )>
+    // clang-format on
+    InIter tag_invoke(hpx::for_each_n_t, InIter first, Size count, F&& f)
+    {
+        static_assert((hpx::traits::is_forward_iterator<InIter>::value),
```

Comment:
For the sequential version, `is_input_iterator` is sufficient (I know we're not consistent on this one, but since we touch it, we might make it proper as we go).